hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/BucketHandles.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client;
+
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.VisibleForTesting;
+import org.apache.hudi.io.HoodieWriteHandle;
+import org.apache.hudi.io.MiniBatchHandle;
+
+import org.apache.hadoop.fs.Path;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is responsible for keeping write handles between checkpoints.
+ * It keeps paths instead of closed handles to reduce memory footprint.
+ * These write paths are used to create ReplaceHandle.
+ */
+public final class BucketHandles {
+  private final Map<String, MiniBatchHandle> fileToHandle;
+  private final Map<String, Path> fileToPath;
+
```

Comment:
Think we can add a `HoodieTable` attribute here, and share this table object in handles.

I can raise a PR to your branch. What do you think?

Reply:
Thank you for helping, I'm reworking the PR now according to https://github.com/apache/hudi/pull/7626#discussion_r1066607740 and `BucketHandles` is going to be removed at all